code/lib/cli/src/generators/configure.ts

```diff
@@ -94,42 +96,70 @@ export async function configureMain({
     .replace('<<import>>', `${imports.join('\n\n')}\n`)
     .replace('<<type>>', isTypescript ? ': StorybookConfig' : '')
     .replace('<<mainContents>>', mainContents);
-  await fse.writeFile(
-    `./${storybookConfigFolder}/main.${isTypescript ? 'ts' : 'js'}`,
-    dedent(mainJsContents),
-    { encoding: 'utf8' }
-  );
+
+  const mainPath = `./${storybookConfigFolder}/main.${isTypescript ? 'ts' : 'js'}`;
+  const prettyMain = prettier.format(dedent(mainJsContents), {
+    ...prettier.resolveConfig.sync(process.cwd()),
+    filepath: mainPath,
+  });
+  await fse.writeFile(mainPath, prettyMain, { encoding: 'utf8' });
 }
 
 export async function configurePreview(options: ConfigurePreviewOptions) {
-  const { prefix = '' } = options.frameworkPreviewParts || {};
+  const { prefix: frameworkPrefix = '' } = options.frameworkPreviewParts || {};
   const isTypescript =
     options.language === SupportedLanguage.TYPESCRIPT_4_9 ||
     options.language === SupportedLanguage.TYPESCRIPT_3_8;
 
+  const rendererPackage =
```

Comment:
Could you add a comment here explaining why there's a filtering for community frameworks?